diff --git a/CMakeLists.txt b/CMakeLists.txt
index cf35bbb..625dc67 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -237,6 +237,12 @@ if( HAVE_LIBM )
    list( APPEND CMAKE_REQUIRED_LIBRARIES -lm )
 endif()
 
+# mingw fails to find math library (used in system libraries), force it here
+if( MINGW )
+    set( LIB_m m )
+    MARK_AS_ADVANCED( FORCE LIB_m )
+endif()
+
 check_library_exists( atomic __atomic_fetch_add_4 "" HAVE_LIBATOMIC )
 if( HAVE_LIBATOMIC )
   list( APPEND CMAKE_REQUIRED_LIBRARIES -latomic )
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 8e31f03..4badeeb 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1064,6 +1064,11 @@ list( APPEND LIBRARIES
       $<$<PLATFORM_ID:Linux,FreeBSD,OpenBSD,NetBSD,CYGWIN>:pthread>
 )
 
+# mingw needs these too
+if( CMAKE_SYSTEM_NAME MATCHES "Windows" AND NOT MSVC)
+    list( APPEND LIBRARIES PUBLIC z pthread)
+endif()
+
 set( BUILDING_AUDACITY YES )
 set( INSTALL_PREFIX "${_PREFIX}" )
 set( PKGLIBDIR "${_PKGLIBDIR}" )
