diff --git a/Lib/distutils/unixccompiler.py b/Lib/distutils/unixccompiler.py
index f159586..004e89f 100644
--- a/Lib/distutils/unixccompiler.py
+++ b/Lib/distutils/unixccompiler.py
@@ -200,6 +200,12 @@ class UnixCCompiler(CCompiler):
 
                 if sys.platform == 'darwin':
                     linker = _osx_support.compiler_fixup(linker, ld_args)
+                elif target_lang == "c++" and self.compiler_cxx:
+                    ld_args += linker[i+offset+1:]
+                    linker = [linker[i+offset]]
+                else:
+                    ld_args += linker[1:]
+                    linker = [linker[0]]
 
                 self.spawn(linker + ld_args)
             except DistutilsExecError as msg:
