diff --git a/src/serd_config.h b/src/serd_config.h
index f9d6269..7e6715c 100644
--- a/src/serd_config.h
+++ b/src/serd_config.h
@@ -33,7 +33,7 @@

 // POSIX.1-2001: fileno()
 #  ifndef HAVE_FILENO
-#    if defined(_POSIX_VERSION) && _POSIX_VERSION >= 200112L
+#    if defined(_POSIX_VERSION) && _POSIX_VERSION >= 200112L && !defined(__EMSCRIPTEN__)
 #      define HAVE_FILENO
 #    endif
 #  endif
@@ -49,7 +49,7 @@

 // POSIX.1-2001: posix_memalign()
 #  ifndef HAVE_POSIX_MEMALIGN
-#    if defined(_POSIX_VERSION) && _POSIX_VERSION >= 200112L
+#    if defined(_POSIX_VERSION) && _POSIX_VERSION >= 200112L && !defined(__EMSCRIPTEN__)
 #      define HAVE_POSIX_MEMALIGN
 #    endif
 #  endif
