diff --git a/src/qjackctlMainForm.cpp b/src/qjackctlMainForm.cpp
index ddc448b..3f4814d 100644
--- a/src/qjackctlMainForm.cpp
+++ b/src/qjackctlMainForm.cpp
@@ -2645,9 +2645,6 @@ void qjackctlMainForm::exitNotifyEvent (void)
 		jackFinished();
 		break;
 	case QProcess::Crashed:
-	#if defined(__WIN32__) || defined(_WIN32) || defined(WIN32)
-		if (!m_bJackStopped)
-	#endif
 		appendMessagesColor(tr("JACK has crashed."), "#cc3366");
 		break;
 	case QProcess::Timedout:
