diff --git a/configure b/configure
index 5af7e64..344ac2d 100755
--- a/configure
+++ b/configure
@@ -5670,7 +5670,7 @@ $as_echo "yes" >&6; }
 fi
 fi
 if test "x$ac_portaudio" = "xyes"; then
-   $as_echo "#define CONFIG_PORTAUDIO 1 Define if PORTAUDIO library is available." >>confdefs.h
+   $as_echo "#define CONFIG_PORTAUDIO 1" >>confdefs.h
 
    ac_cflags="$ac_cflags $PORTAUDIO_CFLAGS"
    ac_libs="$ac_libs $PORTAUDIO_LIBS"
diff --git a/configure.ac b/configure.ac
index a7263e3..52eef13 100644
--- a/configure.ac
+++ b/configure.ac
@@ -402,7 +402,7 @@ if test "x$ac_portaudio" = "xyes"; then
    PKG_CHECK_MODULES([PORTAUDIO], [portaudio-2.0], [ac_portaudio="yes"], [ac_portaudio="no"])
 fi
 if test "x$ac_portaudio" = "xyes"; then
-   AC_DEFINE(CONFIG_PORTAUDIO 1, [Define if PORTAUDIO library is available.])
+   AC_DEFINE(CONFIG_PORTAUDIO, 1, [Define if PORTAUDIO library is available.])
    ac_cflags="$ac_cflags $PORTAUDIO_CFLAGS"
    ac_libs="$ac_libs $PORTAUDIO_LIBS"
 else
diff --git a/src/config.h.in b/src/config.h.in
index 579aa21..a1aedfc 100644
--- a/src/config.h.in
+++ b/src/config.h.in
@@ -66,6 +66,9 @@
 /* Default man page path. */
 #undef CONFIG_MANDIR
 
+/* Define if PORTAUDIO library is available. */
+#undef CONFIG_PORTAUDIO
+
 /* Default installation prefix. */
 #undef CONFIG_PREFIX
 
