diff --git a/src/qjackctlMainForm.cpp b/src/qjackctlMainForm.cpp
index 2e251d6..6dd57af 100644
--- a/src/qjackctlMainForm.cpp
+++ b/src/qjackctlMainForm.cpp
@@ -1327,7 +1327,8 @@ void qjackctlMainForm::startJack (void)
 		const QString sPath = QString::fromUtf8(::getenv("PATH"));
 		QStringList paths = sPath.split(chPathSep);
 	#if defined(__WIN32__) || defined(_WIN32) || defined(WIN32)
-		paths = paths << "C:\\Program Files\\Jack" << "C:\\Program Files (x86)\\Jack";
+		paths = paths << QString("%1\\JACK2").arg(getenv("PROGRAMFILES"))
+		              << QString("%1\\JACK2").arg(getenv("PROGRAMFILES(x86)"));
 	#endif
         #if defined(__APPLE__)
                 paths = paths << "/usr/local/bin/";
