diff --git a/configure b/configure
index 87d9990..0e8c97f 100755
--- a/configure
+++ b/configure
@@ -19443,6 +19443,7 @@ LIBS="$LIBS $pkg_LIBS"
 
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
+#include <stdio.h>
 
 int
 main ()
@@ -19552,6 +19553,7 @@ LIBS="$LIBS $pkg_LIBS"
 
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
+#include <stdio.h>
 
 int
 main ()
@@ -19659,6 +19661,7 @@ LIBS="$LIBS $pkg_LIBS"
 
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
+#include <stdio.h>
 
 int
 main ()
@@ -19772,6 +19775,7 @@ LIBS="$LIBS $pkg_LIBS"
 
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
+#include <stdio.h>
 
 int
 main ()
@@ -19877,6 +19881,7 @@ LIBS="$LIBS $pkg_LIBS"
 
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
+#include <stdio.h>
 
 int
 main ()
@@ -20021,6 +20026,7 @@ LIBS="$LIBS $pkg_LIBS"
 
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
+#include <stdio.h>
 
 int
 main ()
