diff --git a/src/sfconfig.h b/src/sfconfig.h
index 0f30855..a36477b 100644
--- a/src/sfconfig.h
+++ b/src/sfconfig.h
@@ -113,7 +113,11 @@
 #define HAVE_X86INTRIN_H 0
 #endif
 
-#define CPU_IS_X86		(defined __i486__ || defined __i586__ || defined __i686__ || defined __x86_64__)
+#if (defined __i486__ || defined __i586__ || defined __i686__ || defined __x86_64__)
+#define CPU_IS_X86 1
+#else
+#define CPU_IS_X86 0
+#endif
 #define CPU_IS_X86_64	(defined __x86_64__)
 
 #endif
