diff --git a/windows/portaudio/JackPortAudioDriver.h b/windows/portaudio/JackPortAudioDriver.h
index 9897f60..22e3bc1 100644
--- a/windows/portaudio/JackPortAudioDriver.h
+++ b/windows/portaudio/JackPortAudioDriver.h
@@ -22,7 +22,10 @@ Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
 
 #include "JackAudioDriver.h"
 #include "JackPortAudioDevices.h"
+
+#ifdef _WIN32
 #include "JackMMCSS.h"
+#endif
 
 namespace Jack
 {
@@ -31,7 +34,11 @@ namespace Jack
 \brief The PortAudio driver.
 */
 
-class JackPortAudioDriver : public JackMMCSS, public JackAudioDriver
+class JackPortAudioDriver :
+#ifdef _WIN32
+    public JackMMCSS,
+#endif
+    public JackAudioDriver
 {
 
     private:
@@ -55,8 +62,11 @@ class JackPortAudioDriver : public JackMMCSS, public JackAudioDriver
 
     public:
 
-        JackPortAudioDriver(const char* name, const char* alias, JackLockedEngine* engine, JackSynchro* table, PortAudioDevices* pa_devices)
-                : JackMMCSS(), JackAudioDriver(name, alias, engine, table), fStream(NULL), fInputBuffer(NULL), fOutputBuffer(NULL),
+        JackPortAudioDriver(const char* name, const char* alias, JackLockedEngine* engine, JackSynchro* table, PortAudioDevices* pa_devices) :
+#ifdef _WIN32
+                JackMMCSS(),
+#endif
+                JackAudioDriver(name, alias, engine, table), fStream(NULL), fInputBuffer(NULL), fOutputBuffer(NULL),
                 fInputDevice(paNoDevice), fOutputDevice(paNoDevice), fPaDevices(pa_devices)
         {}
 
diff --git a/wscript b/wscript
index 86eb395..89a0b2a 100644
--- a/wscript
+++ b/wscript
@@ -151,7 +151,6 @@ def options(opt):
             'portaudio',
             help='Enable Portaudio driver',
             conf_dest='BUILD_DRIVER_PORTAUDIO')
-    portaudio.check(header_name='windows.h')  # only build portaudio on windows
     portaudio.check_cfg(
             package='portaudio-2.0 >= 19',
             uselib_store='PORTAUDIO',
