diff --git a/common/JackControlAPI.cpp b/common/JackControlAPI.cpp
index c13640a..0e00535 100644
--- a/common/JackControlAPI.cpp
+++ b/common/JackControlAPI.cpp
@@ -458,10 +458,7 @@ jackctl_internals_load(
 
     descriptor_node_ptr = jack_internals_load(NULL);
     if (descriptor_node_ptr == NULL)
-    {
-        jack_error("Could not find any internals in driver directory!");
         return false;
-    }
 
     while (descriptor_node_ptr != NULL)
     {
diff --git a/common/JackDriverLoader.cpp b/common/JackDriverLoader.cpp
index 7715043..9a1be51 100644
--- a/common/JackDriverLoader.cpp
+++ b/common/JackDriverLoader.cpp
@@ -670,10 +670,6 @@ JSList* jack_internals_load(JSList * internals)
 
     } while (FindNextFileW(file, &filedata));
 
-    if (!driver_list) {
-        jack_error ("Could not find any internals in %s!", driver_dir);
-    }
-
  error:
     if (driver_dir) {
         free(driver_dir);
@@ -738,11 +734,6 @@ JSList* jack_internals_load(JSList * internals)
                     driver_dir, strerror (errno));
     }
 
-    if (!driver_list) {
-        jack_error ("Could not find any internals in %s!", driver_dir);
-        return NULL;
-    }
-
     return driver_list;
 }
 
