diff --git a/common/JackInternalClient.cpp b/common/JackInternalClient.cpp
index 6b4dd75..a428c97 100644
--- a/common/JackInternalClient.cpp
+++ b/common/JackInternalClient.cpp
@@ -73,6 +73,10 @@ int JackInternalClient::Open(const char* server_name, const char* name, jack_uui
     int result;
     jack_log("JackInternalClient::Open name = %s", name);
     
+    // ignore client from alsa-plugins
+    if (strncmp(name, "alsa-jack.", 10) == 0)
+        return -1;
+
     if (strlen(name) >= JACK_CLIENT_NAME_SIZE) {
         jack_error("\"%s\" is too long to be used as a JACK client name.\n"
                    "Please use %lu characters or less",
