--- fluidsynth-static-1.1.11.orig/src/drivers/fluid_dsound.c
+++ fluidsynth-static-1.1.11/src/drivers/fluid_dsound.c
@@ -23,7 +23,7 @@
 #include "fluid_adriver.h"
 #include "fluid_settings.h"

-#if DSOUND_SUPPORT
+#if DSOUND_SUPPORT_OFF

 #define INITGUID

--- fluidsynth-static-1.1.11.orig/src/drivers/fluid_adriver.h
+++ fluidsynth-static-1.1.11/src/drivers/fluid_adriver.h
@@ -74,7 +74,7 @@ void delete_fluid_core_audio_driver(fluid_audio_driver_t *p);
 void fluid_core_audio_driver_settings(fluid_settings_t *settings);
 #endif

-#if DSOUND_SUPPORT
+#if DSOUND_SUPPORT_OFF
 fluid_audio_driver_t *new_fluid_dsound_audio_driver(fluid_settings_t *settings,
         fluid_synth_t *synth);
 void delete_fluid_dsound_audio_driver(fluid_audio_driver_t *p);

--- fluidsynth-static-1.1.11.orig/src/drivers/fluid_adriver.c
+++ fluidsynth-static-1.1.11/src/drivers/fluid_adriver.c
@@ -89,7 +89,7 @@ static const fluid_audriver_definition_t fluid_audio_drivers[$
     },
 #endif

-#if DSOUND_SUPPORT
+#if DSOUND_SUPPORT_OFF
     {
         "dsound",
         new_fluid_dsound_audio_driver,
@@ -181,7 +181,7 @@ void fluid_audio_driver_settings(fluid_settings_t *settings)
     fluid_settings_register_str(settings, "audio.driver", "oss", 0);
 #elif COREAUDIO_SUPPORT
     fluid_settings_register_str(settings, "audio.driver", "coreaudio", 0);
-#elif DSOUND_SUPPORT
+#elif DSOUND_SUPPORT_OFF
     fluid_settings_register_str(settings, "audio.driver", "dsound", 0);
 #elif SNDMAN_SUPPORT
     fluid_settings_register_str(settings, "audio.driver", "sndman", 0);
@@ -208,7 +208,7 @@ void fluid_audio_driver_settings(fluid_settings_t *settings)
 #if COREAUDIO_SUPPORT
     fluid_settings_add_option(settings, "audio.driver", "coreaudio");
 #endif
-#if DSOUND_SUPPORT
+#if DSOUND_SUPPORT_OFF
     fluid_settings_add_option(settings, "audio.driver", "dsound");
 #endif
 #if SNDMAN_SUPPORT


--- fluidsynth-static-1.1.11.orig/src/drivers/fluid_aufile.c
+++ fluidsynth-static-1.1.11/src/drivers/fluid_aufile.c
@@ -98,7 +98,7 @@ new_fluid_file_audio_driver(fluid_settings_t *settings,
     return (fluid_audio_driver_t *) dev;

 error_recovery:
-    delete_fluid_file_audio_driver((fluid_audio_driver_t *) dev);
+    delete_fluid_file_audio_driver(fluid_audio_driver_t *);
     return NULL;
 }

