diff --git a/configure.py b/configure.py
index d893ae8..7d13747 100644
--- a/configure.py
+++ b/configure.py
@@ -1558,7 +1558,7 @@ def generate_makefiles(target_config, verbose, parts, tracing, fatal_warnings, d
     inf = open_for_writing('__init__.py')
     inf.write(contents)
 
-    if target_config.py_platform == 'win32':
+    if target_config.py_platform == 'win32' or True:
         # On Windows we try and make sure the Qt DLLs can be found, either any
         # bundled copies or an existing installation (using the traditional
         # Windows DLL search).  We don't raise an exception in case the
@@ -2124,11 +2124,8 @@ def run_make(target_config, verbose, exe, makefile_name):
 
     # Guess the name of make and set the default target and platform specific
     # name of the executable.
-    if target_config.py_platform == 'win32':
-        if target_config.qmake_spec == 'win32-g++':
-            make = 'mingw32-make'
-        else:
-            make = 'nmake'
+    if target_config.py_platform == 'win32' or True:
+        make = 'make'
 
         if target_config.debug:
             makefile_target = 'debug'
@@ -2421,6 +2418,10 @@ def compile_test_program(target_config, verbose, mname, source=None, debug=None)
     pro_lines.append('TARGET = %s' % name)
 
     pro_lines.append('SOURCES = %s' % qmake_quote(name_source))
+    pro_lines.append('QMAKE_CXXFLAGS += $$(CXXFLAGS)')
+    pro_lines.append('QMAKE_CFLAGS += $$(CFLAGS)')
+    pro_lines.append('QMAKE_LFLAGS += $$(LDFLAGS)')
+
 
     f = open_for_writing(name_pro)
     f.write('\n'.join(pro_lines))
@@ -2442,7 +2443,13 @@ def run_test_program(mname, test, verbose):
 
     # Create the output file, first making sure it doesn't exist.
     remove_file(out_file)
-    run_command(test + ' ' + out_file, verbose)
+
+    exe_wrapper = os.getenv('EXE_WRAPPER')
+    if exe_wrapper is not None:
+        exe_wrapper += ' '
+    else:
+        exe_wrapper = ''
+    run_command(exe_wrapper + test + ' ' + out_file, verbose)
 
     if not os.access(out_file, os.F_OK):
         error("%s failed to create %s. Make sure your Qt installation is correct." % (test, out_file))
